pyproject.toml

```
@@ -1,34 +1,184 @@
+]
+build-backend = "hatchling.build"
+
+[project]
+name = "semver"
+description = "Python helper for Semantic Versioning (https://semver.org)"
+readme = "README.rst"
+requires-python = ">=3.7"
+authors = [
+    { name = "Kostiantyn Rybnikov", email = "k-bx@k-bx.com" },
+]
+maintainers = [
+    { name = "Sebastien Celles", email = "s.celles@gmail.com" },
+    { name = "Tom Schraitle" },
+]
+keywords = [
+    "python",
+    "version",
+    "semver",
+    "versioning",
+    "version",
+    "semantic-versioning",
+    "release",
+    "semver-format",
+    "semver-tag",
+    "semver-release",
+    "semver-cli",
+]
+classifiers = [
+    # I think python-semver is more "Console" than "Web Environment"
+    #"Environment :: Web Environment",
+    "Environment :: Console",
+    "Development Status :: 5 - Production/Stable",
+    "Topic :: Software Development",
+    "Intended Audience :: Developers",
+    "License :: OSI Approved :: BSD License",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Topic :: Software Development :: Libraries :: Python Modules",
+]
+dependencies = []
+dynamic = [
+    "version",
+]
+
+[project.urls]
+Homepage = "https://github.com/python-semver/python-semver"
+Changelog = "https://python-semver.readthedocs.io/en/latest/changelog.html"
+Documentation = "https://python-semver.rtfd.io"
+Releases = "https://github.com/python-semver/python-semver/releases"
+Issues = "https://github.com/python-semver/python-semver/issues"
+
+[project.license]
+file = "LICENSE.txt"
+
+[project.scripts]
+pysemver = "semver.cli:main"
+
+[tool.hatch.version]
+path = "src/semver/__about__.py"
+
+[tool.hatch.build]
+include = [
+    "src/semver/*.py",
+    "src/semver/py.typed",
+]
+
+[tool.hatch.envs.default]
+dependencies = [
+    "towncrier",
+    "wheel",
+]
+
+[tool.hatch.envs.style]
+dependencies = [
+    "black",
+    "isort",
+    "flake8",
+    "pycodestyle",
+]
+
+[tool.hatch.envs.style.scripts]
+fmt = [
+    "black .",
+    "isort .",
+]
+lint = [
+    "flake8 --exit-zero",
+    "pycodestyle",
 ]
-build-backend = "setuptools.build_meta"
 
+[tool.hatch.envs.docs]
+dependencies = [
+    "pdoc3"
+]
+
+[tool.hatch.envs.docs.scripts]
+build = "pdoc --html --output-dir api-docs src/semver --force"
+serve = "pdoc --http : python-semver"
+
+[tool.hatch.envs.test]
+dependencies = [
+    "pytest-cov",
+    "tox",
+]
+
+[tool.hatch.envs.test.scripts]
+cov = "pytest -vx"
+no-cov = "cov --no-cov"
+tox_test = "tox"
+
+[[tool.hatch.envs.test.matrix]]
+python = ["310", "311"]
+
+[tool.coverage.run]
+branch = true
+parallel = true
+omit = [
+    # add files to exclude them from the coverage report, e.g.
+    # "src/semver/__about__.py"
+]
```

If I'm not mistaken, you need to tell coverage/pytest-cov where to find the source by using `source`. Furthermore, I would add `semver/_types.py` to the list of files to be excluded from coverage (it contains just type definitions):


```suggestion
[tool.coverage.run]
source = [ "src/semver/*", ]
branch = true
parallel = true
omit = [
    # add files to exclude them from the coverage report, e.g.
    "src/semver/_types.py",
]
```

---

ok, will do.